ui/src/app/views/copy-notebook-modal/component.spec.tsx

```diff
@@ -0,0 +1,119 @@
+import {mount} from 'enzyme';
+import * as React from 'react';
+import Select from 'react-select';
+
+import {registerApiClient, workspacesApi} from 'app/services/swagger-fetch-clients';
+import {WorkspaceAccessLevel, WorkspacesApi} from 'generated/fetch';
+import {WorkspacesApiStub} from 'testing/stubs/workspaces-api-stub';
+import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
+
+import { CopyNotebookModal, CopyNotebookModalProps, CopyNotebookModalState } from './component';
+
+describe('CopyNotebookModal', () => {
+  let props: CopyNotebookModalProps;
+
+  const component = () => {
+    return mount<CopyNotebookModal, CopyNotebookModalProps, CopyNotebookModalState>
+    (<CopyNotebookModal {...props}/>);
+  };
+
+  const workspaces = [
+    {
+      namespace: 'El Capitan',
+      name: 'Freerider',
+      id: 'freerider'
+    },
+    {
+      namespace: 'El Capitan',
+      name: 'Dawn Wall',
+      id: 'dawn wall'
+    },
+    {
+      namespace: 'El Capitan',
+      name: 'Zodiac',
+      id: 'zodiac'
+    },
+    {
+      namespace: 'El Capitan',
+      name: 'The Nose',
+      id: 'the nose'
+    }
+  ];
+
+  beforeEach(() => {
+    props = {
+      fromWorkspaceNamespace: "namespace",
+      fromWorkspaceName: "name",
+      fromNotebook: {
+        name: "notebook",
+        path: "path",
+        lastModifiedTime: null
+      },
+      onClose: () => {},
+      onCopy: () => {}
+    };
+
+    let apiStub = new WorkspacesApiStub(workspaces);
+    registerApiClient(WorkspacesApi, apiStub);
+    apiStub.workspaceAccess.set(workspaces[0].id, WorkspaceAccessLevel.OWNER);
+    apiStub.workspaceAccess.set(workspaces[1].id, WorkspaceAccessLevel.READER);
+    apiStub.workspaceAccess.set(workspaces[2].id, WorkspaceAccessLevel.WRITER);
+    apiStub.workspaceAccess.set(workspaces[3].id, WorkspaceAccessLevel.NOACCESS);
+  });
+
+  it('should render', async() => {
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+  });
+
+  it('should populate select options with writeable Workspaces from getWorkspaces()', async() => {
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+
+    let select = wrapper.find(Select);
+    select.instance().setState({menuIsOpen: true});
+    wrapper.update();
+
+    const options = wrapper.find(Select).find({role: 'option'}).map(e => e.text());
+
+    expect(options).toEqual([workspaces[0].name, workspaces[2].name]);
+  });
+
+  it('should set destination state upon clicking an option', async() => {
```

Comment:
Sure. I  did it this way because I was thinking have more tests enable you to find the root cause of a test failure more quickly but combining them is more future proofed since its testing against its usage rather than the implementation.

Reply:
Since test failures give line number, I am not too worried about combining them. I don't think its complex enough that we as a team are going to have too much trouble finding root cause.